src/Microsoft.PowerShell.Commands.Management/commands/management/GetClipboardCommand.cs

```diff
@@ -211,4 +276,108 @@ private List<PSObject> GetClipboardContentAsFileList()
             return result;
         }
     }
+
+#if !WINFORMS
+    internal static class ClipboardHelper
+    {
+        private static bool? _clipboardSupported;
+
+        private static string StartProcess(
+            string tool,
+            string args,
+            string stdin = "",
+            bool readStdout = true)
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo();
+            startInfo.UseShellExecute = false;
+            startInfo.RedirectStandardInput = true;
+            startInfo.RedirectStandardOutput = true;
+            startInfo.RedirectStandardError = true;
+            startInfo.FileName = tool;
+            startInfo.Arguments = args;
+            string stdout = null;
+
+            using (Process process = new Process())
+            {
+                process.StartInfo = startInfo;
+                try
+                {
+                    process.Start();
+                }
+                catch (System.ComponentModel.Win32Exception)
+                {
+                    _clipboardSupported = false;
+                    throw new PlatformNotSupportedException();
+                }
+
+                if (stdin != string.Empty)
+                {
+                    process.StandardInput.Write(stdin);
+                    process.StandardInput.Close();
+                }
+
+                if (readStdout)
+                {
+                    stdout = process.StandardOutput.ReadToEnd();
+                }
+
+                process.WaitForExit(250);
+
+                _clipboardSupported = process.ExitCode == 0;
+            }
+
+            return stdout;
+        }
+
+        public static string GetText()
+        {
+            if (_clipboardSupported == false)
+            {
+                throw new PlatformNotSupportedException();
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return StartProcess("xclip", "-selection clipboard -out");
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return StartProcess("pbpaste", string.Empty);
+            }
+            else
+            {
+                // Unfortunately there isn't a built in tool to read from
+                // clipboard on Windows if WinForms is not available
```

Comment:
It looks like feature request for Windows API - in PowerShell we could natively set and get any object and we could do this without a lot of extra API from WinForms.

Reply:
There are windows APIs
https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getclipboarddata 
just not DotNet APIs, outside of winforms

Reply:
PSReadLine avoids winforms by using the native Win32 APIs, see [here](https://github.com/PowerShell/PSReadLine/blob/master/PSReadLine/Clipboard.cs).

Reply:
Since the code well-tested in PSReadline we could move it to PowerShell engine and make public for PSReadline.